src/AppInstallerCLICore/Commands/UpgradeCommand.cpp

```diff
@@ -109,17 +159,19 @@ namespace AppInstaller::CLI
     void UpgradeCommand::ValidateArgumentsInternal(Execution::Args& execArgs) const
     {
         if (execArgs.Contains(Execution::Args::Type::Manifest) &&
-            (execArgs.Contains(Execution::Args::Type::Query) ||
-             execArgs.Contains(Execution::Args::Type::Id) ||
-             execArgs.Contains(Execution::Args::Type::Name) ||
-             execArgs.Contains(Execution::Args::Type::Moniker) ||
-             execArgs.Contains(Execution::Args::Type::Version) ||
-             execArgs.Contains(Execution::Args::Type::Channel) ||
-             execArgs.Contains(Execution::Args::Type::Source) ||
-             execArgs.Contains(Execution::Args::Type::Exact) ||
-             execArgs.Contains(Execution::Args::Type::All)))
+            (HasSearchQueryArguments(execArgs) ||
+                HasArgumentsForMultiplePackages(execArgs) ||
+                HasArgumentsForSource(execArgs)))
+        {
+            throw CommandException(Resource::String::BothManifestAndSearchQueryProvided);
+        }
+
+
+        else if (!ShouldListUpgrade(execArgs) 
+                && !HasSearchQueryArguments(execArgs) 
+                && HasArgumentsForInstallOptions(execArgs))
         {
-            throw CommandException(Resource::String::BothManifestAndSearchQueryProvided, "");
+            throw CommandException(Resource::String::InvalidArgumentWithoutQueryError);

```

Comment:
With this check, we are blocking "winget upgrade --all --interactive", is this intended?

Reply:
It wasn't intentional, as now looking at it people would probably like to have the option to make all of their upgrades silent or interactive (although I can't personally imagine having the time to babysit `--all --interactive` :D).

I'll push a fix by tomorrow!

Reply:
Resolved in latest commit!